### PR TITLE
Create v3 of airplane-agents chart

### PR DESCRIPTION
**Pull request by yolken-airplane**
## Description
This change creates a new major version of our airplane-agents Helm chart. There are three backwards-incompatible changes:

1. The `teamID` must be set directly, using a secret for this is no longer supported- the team ID isn't secret (right?) and also having it in a secret breaks some settings related to self-hosted storage.
2. The `airplane.clusterFirstDNS` setting has been removed- this has been superseded by `dnsPolicy`, so it shouldn't be needed anymore.
3. The task `dnsPolicy` defaults to `ClusterFirst` instead of `Default`- this has been a big source of pain for users since `Default` can lead to lots of DNS issues (e.g., not being able to hit other services in the cluster).
4. The default agent lock ID (basically, a way of identifing the agent to the API) has been changed from a random string to a hash of various chart parameters including the namespace, envSlug, and agentLabels- this reduces the risk of conflicts when a chart is updated.

The remainder of the changes should be purely cosmetic:

1. Add comments on all settings
2. Move the more important settings to the top 

cc: @zhan-airplane 

## Testing
Testing to be completed via applying locally.

### CODE REVIEW

File: airplane-agent/values.yaml [hunk 1]
```diff
@@ -1,89 +1,85 @@
...
+  # Whether to enable sending error logs to Airplane. This allows Airplane to better
+  # debug errors and performance regressions in self-hosted agents, and it's recommended
+  # to keep it on.
+  allowRemoteObservability: true
+
+  # The environment to restrict this agent to. Leave blank to allow the agent
+  # to run tasks from all environments.
   envSlug: ""
 
-  apiHost: https://api.airplane.dev
```
Comment (yolken-airplane) [inline airplane-agent/values.yaml:20]: Moved these below.

File: airplane-agent/values.yaml [hunk 2]
```diff
@@ -1,89 +1,85 @@
...
   #
-  # Accepts any of "Default", "ClusterFirst", "ClusterFirstWithHostNet", and "None".
   # See: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy.
-  #
-  # NOTE: If your tasks need to communicate with other services in your cluster, then you probably
-  # want "ClusterFirst" here. We're setting "Default" for backwards compatibility reasons, but this
-  # will be changed in the next major version of this chart.
-  dnsPolicy: Default
-
-  serviceJWTSecret:
```
Comment (yolken-airplane) [inline airplane-agent/values.yaml:50]: Moved this below.

File: airplane-agent/values.yaml [hunk 3]
```diff
@@ -1,89 +1,85 @@
...
-  # Pod and container security contexts for task pods spun up by the agent.
+  # Pod and container security contexts for task runners spun up by the agent.
   taskPodSecurityContext: {}
   taskSecurityContext: {}
 
   # Set to true if you're running inside AWS and want to use ECR-sourced public
   # images. This can improve performance and may reduce networking costs.
   useECRPublicImages: false
 
-  # For development purposes only.
```
Comment (yolken-airplane) [inline airplane-agent/values.yaml:71]: Moved this up.

File: airplane-agent/values.yaml [hunk 4]
```diff
@@ -1,89 +1,85 @@
...
   # Extra environment variables to apply to all agent and task containers. Useful
   # for setting things like HTTP_PROXY.
   extraEnvVars: {}
 
-  # Configuration for remote observability. This allows Airplane to better debug errors
-  # and performance regressions in self-hosted agents.
-  allowRemoteObservability: true
-  otelCollectorURL: https://otel-collector.airplane.dev
-
-  # Whether to log task output to stdout/stderr so it can be picked up
```
Comment (yolken-airplane) [inline airplane-agent/values.yaml:83]: Moved this up.

File: airplane-agent/values.yaml [hunk 5]
```diff
@@ -1,89 +1,85 @@
...
   useECRPublicImages: false
 
-  # For development purposes only.
-  debugMode: false
-
   # Extra environment variables to apply to all agent and task containers. Useful
   # for setting things like HTTP_PROXY.
   extraEnvVars: {}
 
-  # Configuration for remote observability. This allows Airplane to better debug errors
```
Comment (yolken-airplane) [inline airplane-agent/values.yaml:78]: Split these up, moved the boolean up and the URL down.

**Review by ameyakhare — APPROVED** [review]

**Pull request merged**